test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -5,35 +5,242 @@ package scyllacluster
+				},
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}()
+
 		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
-		sc.Spec.Datacenter.Racks[0].Members = 3
 		sc.Spec.AutomaticOrphanedNodeCleanup = true
+		sc.Spec.Datacenter.Racks[0].Members = 3
+		sc.Spec.Datacenter.Racks[0].Storage.StorageClassName = pointer.String(testStorageClass.Name)
+
+		// This test to trigger the orphaned PV cleanup is updating NodeAffinity of PV used by ScyllaCluster to not exiting Node.
+		// The problem is, that local volume storage provisioners which we use in CI, may set this field, which is immutable.
+		// To overcome this, the ScyllaCluster is using custom StorageClass not backed by any provisioner running in the cluster.
+		// ScyllaCluster going to request PVC from that StorageClass, and the test is going to request a clone of the original PVC
+		// from the default StorageClass to get any storage. Then the bound PV is rebounded to the original PVC
+		// but with empty NodeAffinity. This allows the test to trigger the orphaned PV cleanup logic.
+		go func() {
+			defer g.GinkgoRecover()
+
+			lw := &cache.ListWatch{
+				ListFunc: helpers.UncachedListFunc(func(options metav1.ListOptions) (runtime.Object, error) {
+					return f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).List(ctx, options)
+				}),
+				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					return f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).Watch(ctx, options)
+				},
+			}
+			_, _, watcher, done := watchtools.NewIndexerInformerWatcher(lw, &corev1.PersistentVolumeClaim{})
+			defer func() { <-done }()
+			defer watcher.Stop()
+
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case e := <-watcher.ResultChan():
+					switch t := e.Type; t {
+					case watch.Added:
+						pvc := e.Object.(*corev1.PersistentVolumeClaim)
+						if pvc.Spec.StorageClassName == nil || *pvc.Spec.StorageClassName != testStorageClass.Name {
+							continue
+						}
+
+						clone := &corev1.PersistentVolumeClaim{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: fmt.Sprintf("clone-%s", pvc.Name),
+							},
+							Spec: corev1.PersistentVolumeClaimSpec{
+								AccessModes:      pvc.Spec.AccessModes,
+								Selector:         pvc.Spec.Selector,
+								Resources:        pvc.Spec.Resources,
+								VolumeName:       "",
+								StorageClassName: nil,
+								VolumeMode:       pvc.Spec.VolumeMode,
+								DataSource:       pvc.Spec.DataSource,
+								DataSourceRef:    pvc.Spec.DataSourceRef,
+							},
+						}
+
+						framework.Infof("Creating clone PVC for %q", pvc.Name)
+						clone, err = f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).Create(ctx, clone, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						framework.Infof("Creating Pod a clone PVC consumer")
+						consumerPod := makePodSpec(fmt.Sprintf("consumer-%s", clone.Name), image.GetE2EImage(image.BusyBox), clone.Name)
```

Comment:
Could this spec be declared here directly? A named func `makePodSpec` suggests it creates some general purpose pod spec but it runs a container with a hardcoded command and the func is only used here anyway.

Reply:
inlined